[Bug]: OpenJDK does not support agent attachment

### Problem description

The class at https://github.com/openjdk/jdk/blob/jdk-17%2B32/src/jdk.attach/linux/classes/sun/tools/attach/VirtualMachineImpl.java#L49 appears to hardcode the tempdir path to /tmp which is inaccessible on Android without rooting. As a result, JVM attachments fail.

This appears to be the cause of https://github.com/mockito/mockito/issues/2436 on Android

### What steps will reproduce the bug?

Try to initialise a Mockito v5 mock on Android. It will fail to initialise, and interrogating the logs for the underlying ByteBuddy agent yields the following stacktrace:

```
java.lang.reflect.InvocationTargetException
        at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
        at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:77)
        at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
        at java.base/java.lang.reflect.Method.invoke(Method.java:568)                                                                                                                    at net.bytebuddy.agent.Attacher.install(Attacher.java:102)
        at net.bytebuddy.agent.Attacher.main(Attacher.java:64)
Caused by: com.sun.tools.attach.AttachNotSupportedException: Unable to open socket file /proc/11531/root/tmp/.java_pid11531: target process 11531 doesn't respond within 10500ms or HotSpot VM not loaded
        at jdk.attach/sun.tools.attach.VirtualMachineImpl.<init>(VirtualMachineImpl.java:103)
        at jdk.attach/sun.tools.attach.AttachProviderImpl.attachVirtualMachine(AttachProviderImpl.java:58)
        at jdk.attach/com.sun.tools.attach.VirtualMachine.attach(VirtualMachine.java:207)
        ... 6 more
java.lang.reflect.InvocationTargetException
        at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
        at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:77)
        at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
        at java.base/java.lang.reflect.Method.invoke(Method.java:568)                                                                                                                    at net.bytebuddy.agent.Attacher.install(Attacher.java:102)
        at net.bytebuddy.agent.Attacher.main(Attacher.java:64)
Caused by: com.sun.tools.attach.AttachNotSupportedException: Unable to open socket file /proc/11531/root/tmp/.java_pid11531: target process 11531 doesn't respond within 10500ms or HotSpot VM not loaded
        at jdk.attach/sun.tools.attach.VirtualMachineImpl.<init>(VirtualMachineImpl.java:103)
        at jdk.attach/sun.tools.attach.AttachProviderImpl.attachVirtualMachine(AttachProviderImpl.java:58)
        at jdk.attach/com.sun.tools.attach.VirtualMachine.attach(VirtualMachine.java:207)
        ... 6 more
```

### What is the expected behavior?

JVM agent attaches correctly.

### System information

```shell
termux-info
Termux Variables:
TERMUX_APK_RELEASE=F_DROID
TERMUX_APP_PACKAGE_MANAGER=apt
TERMUX_APP_PID=18877
TERMUX_IS_DEBUGGABLE_BUILD=0
TERMUX_MAIN_PACKAGE_FORMAT=debian
TERMUX_VERSION=0.118.0
TERMUX__USER_ID=0
Packages CPU architecture:
aarch64
Subscribed repositories:
# sources.list
deb https://grimler.se/termux/termux-main stable main
Updatable packages:
apt/stable 2.8.1-1 aarch64 [upgradable from: 2.8.1]
cmake/stable 3.30.3 aarch64 [upgradable from: 3.30.2]
command-not-found/stable 2.4.0-40 aarch64 [upgradable from: 2.4.0-37]
coreutils/stable 9.5-1 aarch64 [upgradable from: 9.5]
dpkg/stable 1.22.6-1 aarch64 [upgradable from: 1.22.6]
golang/stable 3:1.23.0-1 aarch64 [upgradable from: 3:1.23.0]
gpgv/stable 2.4.5-2 aarch64 [upgradable from: 2.4.5-1]
libassuan/stable 2.5.7-1 aarch64 [upgradable from: 2.5.7]
libgcrypt/stable 1.10.3-2 aarch64 [upgradable from: 1.10.3-1]
libgpg-error/stable 1.50 aarch64 [upgradable from: 1.49]
libnghttp2/stable 1.63.0 aarch64 [upgradable from: 1.62.1]
libnghttp3/stable 1.5.0 aarch64 [upgradable from: 1.4.0]
libunbound/stable 1.21.0 aarch64 [upgradable from: 1.20.0-1]
nano/stable 8.1 aarch64 [upgradable from: 8.0]
termux-tools/stable 1.43.4-1 all [upgradable from: 1.43.3]
unbound/stable 1.21.0 aarch64 [upgradable from: 1.20.0-1]
termux-tools version:
1.43.3
Android version:
14
Kernel build information:
Linux localhost 5.4.254-qgki-g100549f41f86 #1 SMP PREEMPT Fri Jun 7 17:21:42 CST 2024 aarch64 Android
Device manufacturer:
OnePlus
Device model:
LE2123
LD Variables:
LD_LIBRARY_PATH=
LD_PRELOAD=/data/data/com.termux/files/usr/lib/libtermux-exec.so
Installed termux plugins:
com.termux.styling versionCode:31
```

--- a/src/jdk.attach/linux/classes/sun/tools/attach/VirtualMachineImpl.java
+++ b/src/jdk.attach/linux/classes/sun/tools/attach/VirtualMachineImpl.java
@@ -46,7 +46,7 @@ public class VirtualMachineImpl extends HotSpotVirtualMachine {
     // location is the same for all processes, otherwise the tools
     // will not be able to find all Hotspot processes.
     // Any changes to this needs to be synchronized with HotSpot.
-    private static final String tmpdir = "/tmp";
+    private static final String tmpdir = "@TERMUX_PREFIX@/tmp";
     String socket_path;
     /**
      * Attaches to the target VM

